src/core/colorspace.js

```diff
@@ -696,22 +639,18 @@ class IndexedCS extends ColorSpace {
     this.base = base;
     this.highVal = highVal;
 
-    const baseNumComps = base.numComps;
-    const length = baseNumComps * highVal;
+    const length = base.numComps * highVal;
+    this.lookup = new Uint8Array(length);
 
     if (isStream(lookup)) {
-      this.lookup = new Uint8Array(length);
       const bytes = lookup.getBytes(length);
       this.lookup.set(bytes);
-    } else if (isString(lookup)) {
-      this.lookup = new Uint8Array(length);
+    } else if (typeof lookup === "string") {
       for (let i = 0; i < length; ++i) {
-        this.lookup[i] = lookup.charCodeAt(i);
+        this.lookup[i] = lookup.charCodeAt(i) & 0xff;
```

Comment:
This is consistent with https://github.com/mozilla/pdf.js/blob/38cf49bb8181c4c8b8effef44fbd9d4068fb0d17/src/shared/util.js#L499